protocol/x/clob/keeper/process_operations_test.go

```diff
@@ -1426,6 +1426,289 @@ func TestProcessProposerOperations(t *testing.T) {
 			},
 			expectedError: types.ErrInvalidOrderRemoval,
 		},
+		"Fails with order removal for market in final settlement": {
+			perpetuals: []*perptypes.Perpetual{
+				&constants.BtcUsd_100PercentMarginRequirement,
+			},
+			perpetualFeeParams: &constants.PerpetualFeeParams,
+			clobPairs: []types.ClobPair{
+				constants.ClobPair_Btc_Final_Settlement,
+			},
+			rawOperations: []types.OperationRaw{
+				clobtest.NewOrderRemovalOperationRaw(
+					constants.LongTermOrder_Bob_Num0_Id0_Clob0_Buy25_Price30_GTBT10.OrderId,
+					types.OrderRemoval_REMOVAL_REASON_INVALID_SELF_TRADE,
+				),
+			},
+			expectedError: types.ErrOperationConflictsWithClobPairStatus,
+		},
+		"Fails with short-term order placement for market in final settlement": {
+			perpetuals: []*perptypes.Perpetual{
+				&constants.BtcUsd_100PercentMarginRequirement,
+			},
+			perpetualFeeParams: &constants.PerpetualFeeParams,
+			clobPairs: []types.ClobPair{
+				constants.ClobPair_Btc_Final_Settlement,
+			},
+			rawOperations: []types.OperationRaw{
+				clobtest.NewShortTermOrderPlacementOperationRaw(
+					constants.Order_Dave_Num0_Id1_Clob0_Sell025BTC_Price50000_GTB11,
+				),
+			},
+			expectedError: types.ErrOperationConflictsWithClobPairStatus,
+		},
+		"Fails with ClobMatch_MatchOrders for market in final settlement": {
+			perpetuals: []*perptypes.Perpetual{
+				&constants.BtcUsd_100PercentMarginRequirement,
+			},
+			perpetualFeeParams: &constants.PerpetualFeeParams,
+			clobPairs: []types.ClobPair{
+				constants.ClobPair_Btc_Final_Settlement,
+			},
+			rawOperations: []types.OperationRaw{
+				clobtest.NewMatchOperationRaw(
+					&constants.LongTermOrder_Bob_Num0_Id0_Clob0_Buy25_Price30_GTBT10,
+					[]types.MakerFill{
+						{
+							FillAmount: 10,
+							MakerOrderId: constants.LongTermOrder_Alice_Num0_Id1_Clob0_Sell20_Price10_GTBT10.OrderId,
+						},
+					},
+				),
+			},
+			expectedError: types.ErrOperationConflictsWithClobPairStatus,
+		},
+		// Liquidations are disallowed for markets in final settlement because they may result
+		// in a position increasing in size. This is not allowed for markets in final settlement.
```

Comment:
for my understanding - total OI for the market is going to either stay the same or decrease, right? so technically we can allow liquidations to go through. probably just a product decision?

Reply:
I can ask a product person about this... my thinking was that it is not good to have anyone's position size increasing after the market enters final settlement. If we wanted to allow for liquidations then we could also consider not cancelling stateful orders to increase the chances that the liquidation matches with a maker

Reply:
also we might cancel all short-term orders at some point in the future, so a liquidation order wouldn't even be possible to fill at that point. Don't think we should bother supporting this.

Also if we did support this it would mean we would need to allow short term order placements in the ops queue as well to allow the liquidation order to match against short-term makers. Don't think this is worth the complexity, nor do I think it is helpful in the first place

Reply:
yeah I was just curious. probably not a big deal like you said.

no need to ping product since they were also part of the spec review. as long as you are comfortable with this design decision, let's stick to it 🔥